frictionless_ckan_mapper/ckan_to_frictionless.py

```diff
@@ -150,14 +150,49 @@ def dataset(ckandict):
     # package_show can have license_id and license_title
     # TODO: do we always license_id i.e. can we have license_title w/o
     # license_id?
-    if ('licenses' not in outdict and 'license_id' in outdict):
```

Comment:
The logic below is different: this test said: only do this if `licenses` not already in outdict. You've removed that test and hence you have to do all the work on licenses again below. This isn't the logic we wanted.

Reply:
I guess there could be an improvement over what we had (which was taking existing licenses in extras) which was updating first item in `licenses` array (if already) with stuff at root of ckan dict with values from `license_id`, `license_title` etc. I still think that can then be cleaner.

Reply:
Yes, I have adapted it slightly because "license_id" is supposedly optional according to the docs (I think that's what you meant?).

Reply:
that's a separate point - though i think pretty much license_id is always there but strictly ys it could be optional. the for loop stuff is a good idea ... and the algorithm is then:

1.  Use extras first
2. Updating first item in licenses array (if already there - or create it as empty) with stuff at root of ckan dict i.e. values from license_id, license_title etc.